core/store/ledgerstore/ledger_store.go

```diff
@@ -549,46 +613,106 @@ func (this *LedgerStoreImp) saveBlockToBlockStore(block *types.Block) error {
 	return nil
 }
 
-func (this *LedgerStoreImp) saveBlockToStateStore(block *types.Block) error {
-	blockHash := block.Hash()
-	blockHeight := block.Header.Height
-
+func (this *LedgerStoreImp) executeBlock(block *types.Block) (result store.ExecuteResult, err error) {
 	overlay := this.stateStore.NewOverlayDB()
-
 	if block.Header.Height != 0 {
 		config := &smartcontract.Config{
 			Time:   block.Header.Timestamp,
 			Height: block.Header.Height,
 			Tx:     &types.Transaction{},
 		}
 
-		if err := refreshGlobalParam(config, storage.NewCacheDB(this.stateStore.NewOverlayDB()), this); err != nil {
-			return err
+		err = refreshGlobalParam(config, storage.NewCacheDB(this.stateStore.NewOverlayDB()), this)
+		if err != nil {
+			return
 		}
 	}
 
 	cache := storage.NewCacheDB(overlay)
 	for _, tx := range block.Transactions {
 		cache.Reset()
-		err := this.handleTransaction(overlay, cache, block, tx)
+		notify, e := this.handleTransaction(overlay, cache, block, tx)
+		if e != nil {
+			err = e
+			return
+		}
+
+		result.Notify = append(result.Notify, notify)
+	}
+
+	result.Hash = overlay.ChangeHash()
+	result.WriteSet = overlay.GetWriteSet()
+	if block.Header.Height < STATE_HASH_HEIGHT {
+		result.MerkleRoot = common.UINT256_EMPTY
+	} else if block.Header.Height == STATE_HASH_HEIGHT {
```

Comment:
move the following {} to one separate function?

Reply:
done